tests/Doctrine/Tests/OrmFunctionalTestCase.php

```diff
@@ -343,15 +343,12 @@ abstract class OrmFunctionalTestCase extends OrmTestCase
      */
     final protected function createSchemaForModels(string ...$models): void
     {
-        try {
-            $this->_schemaTool->createSchema(array_map(
-                function (string $className): ClassMetadata {
-                    return $this->_em->getClassMetadata($className);
-                },
-                $models
-            ));
-        } catch (ToolsException $e) {
```

Comment:
@greg0ire Do you know why this was done? It seems that this discards [all kinds of errors](https://github.com/doctrine/orm/runs/5954950927?check_suite_focus=true) in the tests, which sounds terrible to be 🤯 ?

Reply:
Yes I know, I even have written a lengthy commit message when writing that line, please check!

Reply:
@ruudk in case you are having a hard time finding it: https://github.com/doctrine/orm/pull/9526/commits/26e85b8c8868cff5d26c5e1a420e0f4f7fde17b5

Reply:
Ok clear. I reverted my undo commit.

But I'm having a hard time trying to get it to work now.

On 2.11.x it just worked fine.

But somehow it fails on MySQL/MariaDB now. Any advice on how to debug this further?

Reply:
Oke, by doing this https://github.com/doctrine/orm/pull/9636/commits/cb6d10e99a1b734d8c631dceb0b0afe7f3f35ccb  I can see the actual problem:
```
Caused by
Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'CREATE TABLE gh6823_group (id VARCHAR(255) NOT NULL, PRIMARY KEY(id)) DEFAULT CHARACTER SET ascii COLLATE `ascii_unicode_ci` ENGINE = InnoDB':

SQLSTATE[HY000]: General error: 1273 Unknown collation: 'ascii_unicode_ci'
```

Somehow, the table is created with `ascii_unicode_ci` while I specify:
```
 * @Table(name="gh6823_group", options={"charset"="ascii", "collation"="ascii_general_ci"})
 ```

I think there is a bug somewhere in this logic: https://github.com/doctrine/dbal/blob/719663b15983278227669c8595151586a2ff3327/src/Platforms/AbstractMySQLPlatform.php#L459-L462

Will try to debug further

Reply:
Oke, so the problem is that dbal 2.13 doesn't work here... maybe I should just skip this test for dbal 2.13?

Reply:
DBAL 2.13 uses `collate` 🙈 
https://github.com/doctrine/dbal/blob/dc9b3c3c8592c935a6e590441f9abc0f9eba335b/lib/Doctrine/DBAL/Platforms/MySqlPlatform.php#L505-L507

Reply:
Maybe make this fix only for DBAL 3 ? It will me one more incentive to upgrade. DBAL 2 is no longer maintained and will never switch to `collation`.